dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataelement/DefaultCategoryService.java

```diff
@@ -566,11 +566,7 @@ public void generateDefaultDimension()
     @Transactional( readOnly = true )
     public CategoryOptionCombo getDefaultCategoryOptionCombo()
     {
-        CategoryCombo categoryCombo = getCategoryComboByName( CategoryCombo.DEFAULT_CATEGORY_COMBO_NAME );
-
-        return categoryCombo != null && categoryCombo.hasOptionCombos()
-            ? categoryCombo.getOptionCombos().iterator().next()
-            : null;
+        return categoryOptionComboStore.getByName( CategoryCombo.DEFAULT_CATEGORY_COMBO_NAME );
```

Comment:
FYI: I think the old code was fine but in a spring test changes did not become visible unless I would try to reload the value from the database.